pkg/controller/direct/bigtable/mapper.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bigtable
+
+import (
+	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/bigtable/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
+	status "google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	pb "google.golang.org/genproto/googleapis/bigtable/admin/v2"
+)
+
+func BigtableInstanceSpec_FromProto(mapCtx *direct.MapContext, in *pb.Instance) *krm.BigtableInstanceSpec {
+	if in == nil {
+		return nil
+	}
+	out := &krm.BigtableInstanceSpec{}
+	// MISSING: Name
+	out.DisplayName = direct.LazyPtr(in.GetDisplayName())
+	// MISSING: State
+	// MISSING: Type
+	out.InstanceType = direct.Enum_FromProto(mapCtx, in.GetType())
+	// MISSING: Labels
+	// MISSING: CreateTime
+	// MISSING: SatisfiesPzs
+	return out
+}
+func BigtableInstanceSpec_ToProto(mapCtx *direct.MapContext, in *krm.BigtableInstanceSpec) *pb.Instance {
+	if in == nil {
+		return nil
+	}
+	out := &pb.Instance{}
+	// MISSING: Name
+	out.DisplayName = ValueOf(in.DisplayName)
+	out.Type = direct.Enum_ToProto[pb.Instance_Type](mapCtx, in.InstanceType)
+	// MISSING: State
+	// MISSING: Type
+	// MISSING: Labels
+	// MISSING: CreateTime
+	// MISSING: SatisfiesPzs
+	return out
+}
+
+func BackupInfo_StartTime_FromProto(mapCtx *direct.MapContext, in *timestamppb.Timestamp) *string {
+	mapCtx.NotImplemented()
```

Comment:
Do we have a test to ensure that passing a StartTime in a big table resource does not cause an error?

Reply:
We have a fuzz test that checks the reachable functions, yes.  This one is not reachable, because the BackupInfo message is not reachable.  @jingyih is adding reachability testing at the proto level in another PR, so we may be able to not generate these unreachable messages in future (and thus we wouldn't need these methods)

I also think when we have the code to not emit functions that are already defined, we can generate default versions of these functions.  We just don't have enough data yet on what they should look like ... need to do more resources!